core/src/main/java/apoc/cypher/Cypher.java

```diff
@@ -359,3 +265,51 @@ public Stream<MapResult> doWhenCase(
         return whenCase(conditionals, elseQuery, params);
     }
 }
+
+class RunManyResultSpliterator implements Spliterator<Cypher.RowResult>, AutoCloseable {
+    private final Result result;
+    private final long start;
+    private boolean statistics;
+    private int rowCount;
+
+    RunManyResultSpliterator(Result result, boolean statistics) {
+        this.result = result;
+        this.start = System.currentTimeMillis();
+        this.statistics = statistics;
+    }
+
+    @Override
+    public boolean tryAdvance(Consumer<? super Cypher.RowResult> action) {
```

Comment:
We could theoretically get better performance by overriding `forEachRemaining` using `Result.accept(ResultVisitor)`. But I believe procedure framework will traverse the stream one by one anyway so we could not get those benefits without changes there too.